flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/codegen/AsyncCodeGeneratorTest.java

```diff
@@ -0,0 +1,228 @@
+import org.apache.flink.table.utils.CatalogManagerMocks;
+
+import org.apache.flink.shaded.guava31.com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Test for {@link AsyncCodeGenerator}. */
+public class AsyncCodeGeneratorTest {
+
+    private static final RowType INPUT_TYPE =
+            RowType.of(new IntType(), new BigIntType(), new VarCharType());
+
+    private PlannerMocks plannerMocks;
+    private SqlToRexConverter converter;
+
+    private RelDataType tableRowType;
+
+    @BeforeEach
+    public void before() {
+        plannerMocks = PlannerMocks.create();
+        tableRowType =
+                plannerMocks
+                        .getPlannerContext()
+                        .getTypeFactory()
+                        .buildRelNodeRowType(
+                                JavaScalaConversionUtil.toScala(ImmutableList.of("f1", "f2", "f3")),
+                                JavaScalaConversionUtil.toScala(
+                                        ImmutableList.of(
+                                                new IntType(),
+                                                new BigIntType(),
+                                                new VarCharType())));
+        converter =
+                plannerMocks
+                        .getPlanner()
+                        .createToRelContext()
+                        .getCluster()
+                        .getPlanner()
+                        .getContext()
+                        .unwrap(FlinkContext.class)
+                        .getRexFactory()
+                        .createSqlToRexConverter(tableRowType, null);
+
+        plannerMocks
+                .getFunctionCatalog()
+                .registerTemporaryCatalogFunction(
+                        UnresolvedIdentifier.of(
+                                CatalogManagerMocks.DEFAULT_CATALOG,
+                                CatalogManagerMocks.DEFAULT_DATABASE,
+                                "myFunc"),
+                        new AsyncFunc(),
+                        false);
+    }
+
+    @Test
+    public void testStringReturnType() throws Exception {
+        RowData rowData =
+                execute(
+                        "myFunc(f1, f2, f3)",
+                        RowType.of(new VarCharType()),
+                        GenericRowData.of(2, 3L, StringData.fromString("foo")));
+        assertThat(rowData).isEqualTo(GenericRowData.of(StringData.fromString("complete foo 4 6")));
+    }
+
+    @Test
+    public void testTimestampReturnType() throws Exception {
```

Comment:
We don't need tests for different types. This is ensured already through the data structure converter.